android-json-form-wizard/src/main/java/com/vijay/jsonwizard/widgets/CheckBoxFactory.java

```diff
@@ -210,9 +210,14 @@ private void addRequiredValidator(LinearLayout rootLayout, JSONObject jsonObject
             final JSONArray finalCheckBoxValues = checkBoxValues;
             ((Activity) context).runOnUiThread(new Runnable() {
                 @Override
-                public void run(){
+                public void run() {
                     if (StringUtils.isNotEmpty(item.optString(JsonFormConstants.VALUE))) {
-                        checkBox.setChecked(Boolean.parseBoolean(item.optString(JsonFormConstants.VALUE)));
+                        ((JsonApi) context).getAppExecutors().mainThread().execute(new Runnable() {
```

Comment:
Use `formFragment.getJsonApi()` instead. Otherwise, you'll have to add a `context instanceof JsonApi` check which may not be ideal.

Reply:
Done